### PR TITLE
[12.0][FIX] maintenance_plan: plan multicompany rule through equipment

### LINKED ISSUE #135: [12.0] maintenance_plan and muticompany scheme issue
Hello,

I've got a v12 installation of `maintenance_plan` prior to `12.0.2.3.0`. This version adds some new features for plans, like a specific view and menu, which shows plans together.

Plans still haven't `company_id` field, so in this view you can see plans linked to equipments of other companies, and a permission error is raised when accessing to it. And equipment is not a required field for plans.

For this situation, which is the more suitable solution, adding a domain filter for thar view in order to prevent accessing to "forbidden" plans, or directly adding `company_id` field for plans?

cc @etobella @AdriaGForgeFlow

### PR DESCRIPTION
**Pull request by dalonsod**
Fixes #135

### CODE REVIEW

**Comment by dalonsod** [comment]
cc @dreispt @etobella

File: maintenance_plan/models/maintenance_equipment.py [hunk 1]
```diff
@@ -102,10 +102,10 @@ def _cron_generate_requests(self):
             Generates maintenance request on the next_maintenance_date or
             today if none exists
         """
-        for plan in self.env['maintenance.plan'].search(
+        for plan in self.env['maintenance.plan'].sudo().search(
```
Comment (AaronHForgeFlow) [inline maintenance_plan/models/maintenance_equipment.py:105]: but this will do the search as the admin user, so it will search in all companies, right?

Comment (dalonsod) [reply maintenance_plan/models/maintenance_equipment.py:105]: It's right. That code belogs to preventive requests creation task, run by OdooBot user, and this user shouldn't have access to every company, once plan becomes multi-company, so `sudo()` is needed.

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM!

**Review by etobella — APPROVED** [review]
Makes sense

File: maintenance_plan/security/maintenance_plan_security.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<odoo>
+    <data noupdate="1">
+        <record id="maintenance_plan_comp_rule" model="ir.rule">
+            <field name="name">Maintenance Plan Multi-company rule</field>
+            <field name="model_id" ref="model_maintenance_plan"/>
+            <field name="global" eval="True"/>
+            <field name="domain_force">
+                [
+                    '|',
+                    ('equipment_id.company_id','=',False),
```
Comment (etobella) [inline maintenance_plan/security/maintenance_plan_security.xml:11]: Just one thing, in order to do this kind of search, you should check that the field equipment_id has auto_join

Comment (dalonsod) [reply maintenance_plan/security/maintenance_plan_security.xml:11]: @etobella looking at `equipment_id` definition in `maintenance.plan` model, `auto_join` is not defined. I'll add it ASAP.

Comment (dalonsod) [reply maintenance_plan/security/maintenance_plan_security.xml:11]: > @etobella looking at equipment_id definition in maintenance.plan model, auto_join is not defined. I'll add it ASAP.

Done!

**Comment by dalonsod** [comment]
@etobella could you merge this PR? Thanks!

**Pull request merged**